Dockerfile

```diff
@@ -0,0 +1,33 @@
+FROM python:3.6.5-jessie
+
+env DEBIAN_FRONTEND noninteractive
+RUN apt-get update && \
+    apt-get -y install \
+            dbus-x11 \
+            libdbus-1-dev \
+            libespeak1 \
+            libnotify-cil-dev \
+            notification-daemon \
+            sudo \
+            xorg
+
+RUN pip3 install virtualenv
+
+
+COPY . /app
+WORKDIR /app
+
+RUN sudo /bin/bash ./docker.sh
+
+RUN useradd -Um jarvis && echo "jarvis  ALL=(ALL) NOPASSWD:ALL" >> /etc/sudoers
+RUN chown -R jarvis /app
+USER jarvis
+
+RUN sudo /bin/bash ./setup.sh --python3
```

Comment:
It may not be worth the effort, but I think that if you create a docker image for jarvis, it should not use the setup.sh script since environment is well controlled compared to bare metal machine.

Instead, iI suggest you install deps in Dockerfile, and provide several Dockerfiles if you wish to support several distributions (which may not be necessary). This way you will reduce drastically image size (no virtualenv and no cache when installing with apt or pip, remove archives when installing phantomjs, etc..).